WPFSKillTree/Utils/UrlProcessing/BanditConverter.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using POESKillTree.Model;
+
+namespace POESKillTree.Utils.UrlProcessing
+{
+    /// <summary>
+    /// Provides functionality for converting bandits identifiers from external sources into used in the POESKillTree application.
+    /// </summary>
+    public class BanditConverter
```

Comment:
This delegate makes code reusable, but the class in question prevents copy-paste better and adds some degree of compile-time protection.
Since C# compiler is capable of inferring delegate type it makes possible to pass any suitable method or lambda. In this case if someone needs an instance of the BuildUrlData class and is not aware of BanditConverters, it is possible to replicate the same switch/case logic due to ignorance or even worth to write something erroneous like `new BuildUrlData(id => (Bandit)42)` that would fail only in run-time. On the other hand it is much harder to ignore required class.
Therefore, if this class survives other reviews, I would strongly recommend to accept it as is.